### PR TITLE
[7.7.x] RHPAM-1015: Improve computation performance

### PR DESCRIPTION
**Pull request by romartin**
Hey @hasys 

Here are a few improvements taken from Mark's PR (https://github.com/kiegroup/lienzo-core/pull/27) in order to improve the performance.  We'll keep working during next days on more stuff here in lienzo....

This is part of the job for https://issues.jboss.org/browse/RHPAM-1015

Includes https://github.com/kiegroup/kie-wb-common/pull/1813

Thanks!

### CODE REVIEW

File: src/main/java/com/ait/lienzo/client/core/util/Geometry.java [hunk 1]
```diff
@@ -537,6 +516,208 @@ public static final double binomial(final int n, final int k)
...
+            {
+                s = (ix - lx[0]) / (lx[1] - lx[0]);
+            }
+            else
+            {
+                s = (iy - ly[0]) / (ly[1] - ly[0]);
+            }
+
+            // in bounds?
+            //if (t>0 && t<1.0 && s>0 && s<1.0)
```
Comment (hasys) [inline src/main/java/com/ait/lienzo/client/core/util/Geometry.java]: Is question still actual?

File: src/main/java/com/ait/lienzo/client/core/util/Geometry.java [hunk 2]
```diff
@@ -537,6 +516,208 @@ public static final double binomial(final int n, final int k)
...
+        else if (dq > 0) {
+            double rdq = Math.sqrt(dq);
+            roots[0] = (-p[2] - rdq) / (2*p[1]);
+            roots[1] = (-p[2] + rdq) / (2*p[1]);
+        }
+        else
+        {
+            //Console.get().warn("quadraticRoots had dq < 0");
+            // Complex roots
+            //return null;
```
Comment (hasys) [inline src/main/java/com/ait/lienzo/client/core/util/Geometry.java]: Not finished or dead code.

**Comment by mdproctor** [comment]
Please don't merge this. It copy/pastes from my stuff, rather than cherry-pick. The result will be very easy to merge. Also the intersect code, on it's own is not perf related - it would have to be combined with other stuff.

**Comment by romartin** [comment]
Hey @hasys @mdproctor 

I'm agree on not merging it, anyway we're planning to provide fixes here so there is no need to include this yet.

So I'm going to close both PRs about this (the Stunner one too) and just create a new one for the Stunner improvements side, ok? It provides quite some good improvements that we can anyway integrate, and once we've fixes here we'll provide a new PR for those.

Thx

**Pull request closed without merging**